service/app/util/RequestValidator.java

```diff
@@ -1088,4 +1095,49 @@ public static void validateUpdateEvent(Request contentRequestDto) {
       }
     }
   }
+
+  public static Boolean isProgram(String contentId) {
+    Boolean isProgram = false;
+    try {
+      String baseUrl = ProjectUtil.getConfigValue(JsonKey.EKSTEP_BASE_URL);
+      String authorizationToken = getAuthorizationToken();
+
+      Map<String, String> headers = new HashMap<>();
+      headers.put(JsonKey.AUTHORIZATION, JsonKey.BEARER + authorizationToken);
+
+      logger.info(null, "Making call for content read for id: " + contentId);
+      HttpResponse<String> httpResponse = Unirest.get(baseUrl + PropertiesCache.getInstance().getProperty(JsonKey.SUNBIRD_CONTENT_READ_API) + "/" + contentId)
+              .headers(headers)
+              .asString();
+
+      String responseBody = httpResponse.getBody();
+      Map<String, Object> response = mapper.readValue(responseBody, Map.class);
+      Map<String, Object> data = (Map<String, Object>) response.get(JsonKey.RESULT);
+
+      if (org.apache.commons.collections4.MapUtils.isNotEmpty(data)) {
+        Map<String, Object> content = (Map<String, Object>) data.get(JsonKey.CONTENT);
+        String primaryCategory = (String) content.get(JsonKey.PRIMARYCATEGORY);
```

Comment:
you have to use courseCategory not primaryCategory
you need to check cumulativeTracking attribute too -- if that is false -- API should accept state update request...